production-vms/resources.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"production-vms/config"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+)
+
+func createProductionVM(ctx context.Context, vhd *VHD, subnetID string) error {
+	fmt.Printf("Creating VM %s in resource group %s\n", vhd.name, config.ResourceGroupName)
+
+	fmt.Printf("vhd resource: %s\n", vhd)
+
+	vmSize := "Standard_D8ds_v5"
```

Comment:
just food for thought and since you're already working from scratch, maybe consider trying out https://github.com/samber/lo? I was recently made aware of it and have been noticing its usage a bit more in other codebases, seems to make stuff like this a lot cleaner